src/tools/fuzzing.h

```diff
@@ -2110,9 +2110,18 @@ class TranslateToFuzzReader {
         return builder.makeRefNull(type);
       }
       // Last resort: create a function.
+      auto heapType = type.getHeapType();
```

Comment:
So is it guaranteed that `type.isFunction()` by the time we reach here? Can it be data? (Not `dataref`, because it was excluded above, but other data, like array)

If it is guaranteed to be a function somehow, how can its heaptype not be a signature?

Reply:
You're right, the code above would let an externref fall through... which I forgot is not a supertype of function.

Should be fixed now.

The one case where a function is not a signature is funcref. That is a function, but has no signature info.

Reply:
Thanks. This is unrelated, but the comment in line 2080 refering to `func` seems outdated. It should be `funcContext`, right?

Reply:
Yes, good catch, I'll update that.